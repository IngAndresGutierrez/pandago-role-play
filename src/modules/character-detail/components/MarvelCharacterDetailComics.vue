<template>
  <v-container class="py-4">
    <v-row justify="center">
      <v-col cols="12">
        <v-carousel v-if="characterComics.length" height="auto" hide-delimiters>
          <v-carousel-item v-for="(comic, index) in characterComics" :key="index">
            <v-row>
              <v-col cols="12">
                <h3 class="title">{{ comic.title }}</h3>
              </v-col>
            </v-row>
            <v-img :alt="comic.title" height="300px" :src="getImageFromThumbnail(comic.thumbnail)" />
            <v-row>
              <v-col cols="12">
                <h3 v-if="comic.prices.length" class="mt-2 mb-2 text-center">${{ comic.prices[0].price }}</h3>
              </v-col>
            </v-row>
          </v-carousel-item>
        </v-carousel>
        <v-container v-if="loading">
          <v-row justify="center">
            <v-progress-circular color="primary" indeterminate />
          </v-row>
        </v-container>
        <v-alert v-if="!characterComics.length && !loading" type="info">Este personaje no aparece en ningún cómic.</v-alert>
      </v-col>
    </v-row>
  </v-container>
</template>

<script setup lang="ts">
  import { onMounted } from 'vue'
  import { useRoute } from 'vue-router'

  import { Comic } from '../models/comics'
  import marvelService from '../services/character'
  import { getImageFromThumbnail } from '../../common/utils/images'

  const route = useRoute()
  const characterComics = ref<Comic[]>([])
  const loading = ref(false)

  const fetchCharacterDetailComics = async () => {
    loading.value = true

    if (route.params && 'id' in route.params) {
      const characterId = Number(route.params.id)
      characterComics.value = await marvelService.getCharacterDetailComics(characterId)
    }

    loading.value = false
  }

  onMounted(fetchCharacterDetailComics)

</script>

<style scoped>
.title {
  text-align: center;
  padding-bottom: 28px;
}
</style>
