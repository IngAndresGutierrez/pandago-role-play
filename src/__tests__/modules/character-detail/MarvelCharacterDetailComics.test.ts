import 'vuetify/styles'
import { mount } from '@vue/test-utils'
import { createVuetify } from 'vuetify'
import { describe, expect, test } from 'vitest'

import MarvelCharacterDetailComics from '@/modules/character-detail/components/MarvelCharacterDetailComics.vue'

const vuetify = createVuetify()

const mockCharacter = {
  id: 1,
  name: 'Test character',
  thumbnail: {
    path: 'https://mypath.com/images',
    extension: 'jpg',
  },
}

describe('MarvelCharacterDetailComics.vue', () => {
  test('renders correctly', () => {
    const wrapper = mount(MarvelCharacterDetailComics, {
      global: {
        plugins: [vuetify],
      },
      props: {
        character: mockCharacter,
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
