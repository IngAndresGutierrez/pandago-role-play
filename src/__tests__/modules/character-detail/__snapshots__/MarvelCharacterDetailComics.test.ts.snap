// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MarvelCharacterDetailComics.vue > renders correctly 1`] = `
<div
  character="[object Object]"
  class="v-container v-locale--is-ltr py-4"
  data-v-a3f18efc=""
>
  <div
    class="v-row justify-center"
    data-v-a3f18efc=""
  >
    <div
      class="v-col v-col-12"
      data-v-a3f18efc=""
    >
      <!--v-if-->
      <!--v-if-->
      <div
        class="v-alert v-theme--light bg-info v-alert--density-default v-alert--variant-flat"
        data-v-a3f18efc=""
        role="alert"
      >
        
        <!---->
        <span
          class="v-alert__underlay"
        />
        
        <!---->
        <div
          class="v-alert__prepend"
        >
          <i
            aria-hidden="true"
            class="mdi-information mdi v-icon notranslate v-theme--light"
            density="default"
            style="font-size: 28px; height: 28px; width: 28px;"
          />
        </div>
        <div
          class="v-alert__content"
        >
          <!---->
          <!---->
          
          Este personaje no aparece en ningún cómic.
          
        </div>
        <!---->
        <!---->
      </div>
    </div>
  </div>
</div>
`;
